File: problems/0001_two_sum/0001_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        for i, num in enumerate(nums):
            pair = target - num
            if pair in num_map:
                return [num_map[pair], i]
            num_map[num] = i
        return []

File: problems/0001_two_sum/0001_two_sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for i, num := range nums {
        if idx, found := m[target - num]; found {
            return []int{idx, i}
        }
        m[num] = i
    }
    return []int{}
}